packages/shared-internals/tests/colocation.test.ts

```diff
@@ -0,0 +1,25 @@
+import { syntheticJStoHBS } from '../src';
+
+describe('colocation utils', function () {
+  describe('syntheticJStoHBS', function () {
+    test('it returns .hbs files for .js', function () {
+      const testCases = [
+        ['foo.js', 'foo.hbs'],
+        ['foo.js?qp', 'foo.hbs?qp'],
+        ['foo/json.js', 'foo/json.hbs'],
```

Comment:
This is the case that failed before:
![image](https://github.com/user-attachments/assets/0ef3eac9-2edd-459f-aa66-61e1ff94a64c)